test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java

```diff
@@ -166,4 +167,18 @@ public void testRandomUniqueTotallyUnique() {
     public void testRandomUniqueNormalUsageAlwayMoreThanOne() {
         assertThat(randomUnique(() -> randomAlphaOfLengthBetween(1, 20), 10), hasSize(greaterThan(0)));
     }
+
+    public void testRandomValueOtherThan() {
+        // "normal" way of calling where the value is not null
+        int bad = randomInt();
+        assertNotEquals(bad, (int) randomValueOtherThan(bad, ESTestCase::randomInt));
+
+        /*
+         * "funny" way of calling where the value is null. This once
+         * had a unique behavior but at this point `null` acts just
+         * like any other value.
+         */
+        Supplier<Object> usuallyNull = () -> usually() ? null : randomInt();
+        assertNotNull(null, randomValueOtherThan(null, usuallyNull));
```

Comment:
Why pass a null message here, I think `assertNotNull(randomValueOtherThan(null, usuallyNull));` is fine here, the expectation will fail saying we got `null`?

Reply:
My mistake. I originally had `assertThat` and wasn't super careful reworking it. Strange that a null message is allowed though....